pyiron_base/generic/filedata.py

```diff
@@ -83,55 +84,74 @@ def load_file(filename, filetype=None, project=None):
             :class:`pandas.DataFrame`: containing data from file of filetype = '.csv'
 
     """
-    def _load_txt(file):
-        with open(file, encoding='utf8') as f:
-            return f.readlines()
+    def _load_txt(file, is_file):
+        if is_file:
+            with open(file, encoding='utf8') as f:
+                return f.readlines()
+        else:
+            return file.readlines()
 
     def _load_ipynb(file):
         return OwnNotebookNode(nbformat.read(file, as_version=4))
 
-    def _load_json(file):
-        with open(file) as f:
-            return json.load(f)
+    def _load_json(file, is_file):
+        if is_file:
+            with open(file) as f:
+                return json.load(f)
+        else:
+            return json.load(file)
 
     def _load_csv(file):
         return pandas.read_csv(file)
 
     def _load_img(file):
         return Image.open(file)
  
-    def _load_default(file):
+    def _load_default(file, is_file):
         try:
-            return _load_txt(file)
+            return _load_txt(file, is_file)
         except Exception as e:
             raise IOError("File could not be loaded.") from e
 
-    if filetype is None:
-        _, filetype = os.path.splitext(filename)
+    filename_is_str = isinstance(fp, str)
+
+    if filetype is None and filename_is_str:
+        _, filetype = os.path.splitext(fp)
+    elif filetype is None and hasattr(fp, 'name'):
+        _, filetype = os.path.splitext(fp.name)
+    elif filetype is None:
+        return _load_default(fp, filename_is_str)
     elif filetype[0] != '.':
         filetype = '.' + filetype
     filetype = filetype.lower()
 
-    if filetype in ['.h5', '.hdf']:
+    if filetype in ['.h5', '.hdf'] and filename_is_str:
         if project is None:
-            return FileHDFio(file_name=filename)
+            return FileHDFio(file_name=fp)
         else:
-            return ProjectHDFio(file_name=filename, project=project)
+            return ProjectHDFio(file_name=fp, project=project)
     elif filetype in ['.json']:
-        return _load_json(filename)
+        return _load_json(fp, is_file=filename_is_str)
     elif filetype in ['.txt']:
-        return _load_txt(filename)
+        return _load_txt(fp, is_file=filename_is_str)
     elif filetype in ['.csv']:
-        return _load_csv(filename)
+        return _load_csv(fp)
     elif _has_imported['nbformat'] and filetype in ['.ipynb']:
-        return _load_ipynb(filename)
+        return _load_ipynb(fp)
     elif _has_imported['PIL'] and filetype in Image.registered_extensions():
-        return _load_img(filename)
+        return _load_img(fp)
     else:
-        return _load_default(filename)
+        return _load_default(fp, is_file=filename_is_str)
+
+
+class FileDataTemplate(ABC):
+    @property
+    def data(self):
+        """Return the associated data."""
+        raise NotImplementedError
```

Comment:
```suggestion
    @property
    @abstractmethod
    def data(self):
        """Return the associated data."""
        pass  
```
Deriving from `ABC` only checks methods that are marked with `abstractmethod`.